src/snowflake/snowpark/_internal/compiler/utils.py

```diff
@@ -195,6 +198,19 @@ def to_selectable(plan: LogicalPlan, query_generator: QueryGenerator) -> Selecta
         raise ValueError(f"parent type {type(parent)} not supported")
 
 
+def valid_for_replacement(node: LogicalPlan) -> bool:
+    if node._is_valid_for_replacement:
+        return True
+
+    if isinstance(node, SnowflakePlan) and node.source_plan is not None:
+        return node._is_valid_for_replacement
```

Comment:
i am kind of confused about why we need this function here?

Reply:
Can you also add comment for this function?

Reply:
while testing, I noticed that large query breakdown failed with error saying node is not valid for replacement for a `SnowflakePlan` whose child was a `WithQueryBlock`. At the time, I though this function would help us catch-up all such cases but on a second thought, it was a bug in repeated subquery elimination. I fixed the bug and removed this function now.